trie/patriciaMerkleTrie.go

```diff
@@ -247,10 +247,27 @@ func (tr *patriciaMerkleTrie) Recreate(root []byte) (common.Trie, error) {
 	tr.mutOperation.Lock()
 	defer tr.mutOperation.Unlock()
 
-	return tr.recreate(root)
+	return tr.recreate(root, tr.trieStorage)
 }
 
-func (tr *patriciaMerkleTrie) recreate(root []byte) (*patriciaMerkleTrie, error) {
+// RecreateFromEpoch returns a new trie, given the options
+func (tr *patriciaMerkleTrie) RecreateFromEpoch(options common.RootHashHolder) (common.Trie, error) {
+	tr.mutOperation.Lock()
+	defer tr.mutOperation.Unlock()
+
+	if !options.GetEpoch().HasValue {
```

Comment:
might worth some checks on the `options` param?

Reply:
Done.